src/Modules/Manage/Dnn.PersonaBar.Users/Components/UsersController.cs

```diff
@@ -390,65 +381,52 @@ public static bool IsAdmin(PortalSettings portalSettings)
         private static IEnumerable<UserBasicDto> GetUsersFromDb(GetUsersContract usersContract, bool isSuperUser, out int totalRecords)
         {
             totalRecords = 0;
-            List<UserBasicDto> users = null;
-            IEnumerable<UserInfo> userInfos = null;
+            IEnumerable<UserBasicDto> users = null;
 
             var portalId = usersContract.PortalId;
             var pageIndex = usersContract.PageIndex;
             var pageSize = usersContract.PageSize;
-            var paged = false;
 
             switch (usersContract.Filter)
             {
                 case UserFilters.All:
-                    users = GetUsers(usersContract, true, true, isSuperUser, out totalRecords);
-                    paged = true;
+                    users = GetUsers(usersContract, null, null, isSuperUser ? null : (bool?)false, out totalRecords);
                     break;
                 case UserFilters.Authorized:
-                    users = GetUsers(usersContract, false, false, isSuperUser, out totalRecords);
-                    paged = true;
+                    users = GetUsers(usersContract, true, false, isSuperUser ? null : (bool?)false, out totalRecords);
                     break;
                 case UserFilters.SuperUsers:
                     if (isSuperUser)
                     {
-                        var dbUsers = UserController.GetUsers(Null.NullInteger, pageIndex, pageSize, ref totalRecords, true, true);
-                        userInfos = dbUsers?.OfType<UserInfo>().ToList();
+                        users = GetUsers(usersContract, null, null, true, out totalRecords);
                     }
-                    paged = true;
                     break;
                 case UserFilters.UnAuthorized:
-                    users = GetUsers(usersContract, true, false, isSuperUser, out totalRecords);
-                    paged = true;
+                    users = GetUsers(usersContract, false, false, isSuperUser ? null : (bool?)false, out totalRecords);
                     break;
                 case UserFilters.Deleted:
-                    users = GetUsers(usersContract, true, true, isSuperUser, out totalRecords);
-                    paged = true;
+                    users = GetUsers(usersContract, null, true, isSuperUser ? null : (bool?)false, out totalRecords);
                     break;
-                //                    case UserFilters.Online:
-                //                        dbUsers = UserController.GetOnlineUsers(usersContract.PortalId);
-                //                        break;
                 case UserFilters.RegisteredUsers:
```

Comment:
What is the difference between "Authorized" state and member of "Registered Users" role - AFAIK there is none.
The only interesting role I, could think of, would be Administrators role - but there is another option to view its members.

Reply:
THe registered user is a role while the authorized is a state; they are totally independent. I don't know what did it mean in the first place to have users in the Registered Users role; but this is a history and goes way before I got involved with DNN :)

Reply:
AFAIK membership in registeredUsers role is maintained automatically by DNN, when a user gets authorized and unauthorized or deleted